contracts/rewards/src/error.rs

```diff
@@ -0,0 +1,32 @@
+use axelar_wasm_std_derive::IntoContractError;
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq, IntoContractError)]
+pub enum ContractError {
+    #[error("error saving params")]
+    SaveParams,
+
+    #[error("error saving epoch tally")]
+    SaveEpochTally,
+
+    #[error("error saving event")]
+    SaveEvent,
+
+    #[error("error saving rewards pool")]
+    SaveRewardsPool,
+
+    #[error("error saving rewards watermark")]
+    SaveRewardsWatermark,
+
+    #[error("error loading epoch tally")]
+    LoadEpochTally,
+
+    #[error("error loading event")]
+    LoadEvent,
+
+    #[error("error loading rewards pool")]
+    LoadRewardsPool,
+
+    #[error("error loading rewards watermark")]
+    LoadRewardsWatermark,
+}
```

Comment:
I don't really see these as contract level errors, but we don't really have a good example to point you at just yet. I'm ok with keeping it like that for now but we should revisit

Reply:
yeah I wasn't really sure what was best here